### ISSUE TITLE
[Question] problem porting to Baichuan 13B

### ISSUE THREAD

**Issue by charryshi** [haotian-liu/LLaVA#346]
### Question

Thanks to the author. When porting to baichuan 13B, there are a lot of warnings when doing finetuning, but the dataset LLaVA-Instruct-150K seem to right, What could be the problem？ thank you

WARNING: tokenization mismatch: 403 vs. 406. (ignored)
WARNING: tokenization mismatch: 445 vs. 449. (ignored)
WARNING: tokenization mismatch: 406 vs. 410. (ignored)
WARNING: tokenization mismatch: 504 vs. 508. (ignored)
WARNING: tokenization mismatch: 296 vs. 300. (ignored)
WARNING: tokenization mismatch: 257 vs. 261. (ignored)
WARNING: tokenization mismatch: 215 vs. 219. (ignored)
WARNING: tokenization mismatch: 676 vs. 680. (ignored)
WARNING: tokenization mismatch: 145 vs. 148. (ignored)

**Comment by haotian-liu** [comment]
Hi, this may be due to the prompt preprocessing needs modification. [This function](https://github.com/haotian-liu/LLaVA/blob/main/llava/train/train.py#L404) may need refactoring for supporting different prompt modes. For now, we would have to modify this accordingly for prompts that have different structures.

**Comment by charryshi** [comment]
Thanks for reply.

**Comment by dengtianbi** [comment]
> ### Question
> Thanks to the author. When porting to baichuan 13B, there are a lot of warnings when doing finetuning, but the dataset LLaVA-Instruct-150K seem to right, What could be the problem？ thank you
> 
> WARNING: tokenization mismatch: 403 vs. 406. (ignored) WARNING: tokenization mismatch: 445 vs. 449. (ignored) WARNING: tokenization mismatch: 406 vs. 410. (ignored) WARNING: tokenization mismatch: 504 vs. 508. (ignored) WARNING: tokenization mismatch: 296 vs. 300. (ignored) WARNING: tokenization mismatch: 257 vs. 261. (ignored) WARNING: tokenization mismatch: 215 vs. 219. (ignored) WARNING: tokenization mismatch: 676 vs. 680. (ignored) WARNING: tokenization mismatch: 145 vs. 148. (ignored)

Hi,
I encountered the same problem, did you solve it?

**Comment by llpin1992** [comment]
I encountered the same problem, can you give me some guidance how to solve it? @haotian-liu

**Comment by chenhaoran176879** [comment]
did you figure this out? I have trouble porting to baichuan-2 and reading the preprocessing and conversation formation code. @charryshi

**Comment by chenhaoran176879** [comment]
> My solution [csdn](https://blog.csdn.net/qq_37668436/article/details/134253349?csdn_share_tail=%7B%22type%22%3A%22blog%22%2C%22rType%22%3A%22article%22%2C%22rId%22%3A%22134253349%22%2C%22source%22%3A%22qq_37668436%22%7D). This is because the baichuan model didn't use bos token, make tokenizer.decode loss at the begin of the sentence.

Thanks! I will verify the final performance of llava-baichuan-2-13b-chat soon.

**Comment by xmy0916** [comment]
> > My solution [csdn](https://blog.csdn.net/qq_37668436/article/details/134253349?csdn_share_tail=%7B%22type%22%3A%22blog%22%2C%22rType%22%3A%22article%22%2C%22rId%22%3A%22134253349%22%2C%22source%22%3A%22qq_37668436%22%7D). This is because the baichuan model didn't use bos token, make tokenizer.decode loss at the begin of the sentence.
> 
> Thanks! I will verify the final performance of llava-baichuan-2-13b-chat soon.

have you test between llava-baichuan-2-13b-chat and llava-baichuan-2-13b-base?

**Comment by chenhaoran176879** [comment]
> > > My solution [csdn](https://blog.csdn.net/qq_37668436/article/details/134253349?csdn_share_tail=%7B%22type%22%3A%22blog%22%2C%22rType%22%3A%22article%22%2C%22rId%22%3A%22134253349%22%2C%22source%22%3A%22qq_37668436%22%7D). This is because the baichuan model didn't use bos token, make tokenizer.decode loss at the begin of the sentence.
> > 
> > 
> > Thanks! I will verify the final performance of llava-baichuan-2-13b-chat soon.
> 
> have you test between llava-baichuan-2-13b-chat and llava-baichuan-2-13b-base?

Not yet, just chat. Due to previous bug, I have stopped running for several weeks...

**Comment by xmy0916** [comment]
> > > > My solution [csdn](https://blog.csdn.net/qq_37668436/article/details/134253349?csdn_share_tail=%7B%22type%22%3A%22blog%22%2C%22rType%22%3A%22article%22%2C%22rId%22%3A%22134253349%22%2C%22source%22%3A%22qq_37668436%22%7D). This is because the baichuan model didn't use bos token, make tokenizer.decode loss at the begin of the sentence.
> > > 
> > > 
> > > Thanks! I will verify the final performance of llava-baichuan-2-13b-chat soon.
> > 
> > 
> > have you test between llava-baichuan-2-13b-chat and llava-baichuan-2-13b-base?
> 
> Not yet, just chat. Due to previous bug, I have stopped running for several weeks...

i have tested baichuan2-13B, due to the high vram requests(about 80GB/batch size=1) during finetuning, we cannot report the results. results on baichuan2-7B in my private dataset seems worse. Some key metrics droped about 10 points

**Comment by chenhaoran176879** [comment]
quite slow finetuning 13B... 8xA800  need 80 hours or more.
This seems strange since vicuna-13B costed much less.
loss dropped to around 0.8, maybe a week later I shall report.
Thanks for your debug again.

**Comment by xmy0916** [comment]
> quite slow finetuning 13B... 8xA800 need 80 hours or more. This seems strange since vicuna-13B costed much less. loss dropped to around 0.8, maybe a week later I shall report. Thanks for your debug again.

maybe due to the vocub size of baichuan is 4X lager than vicuna.

**Comment by wucx888** [comment]
> quite slow finetuning 13B... 8xA800 need 80 hours or more. This seems strange since vicuna-13B costed much less. loss dropped to around 0.8, maybe a week later I shall report. Thanks for your debug again.

can you share the result on baichuan-13B? thx

**Comment by xmy0916** [comment]
> > quite slow finetuning 13B... 8xA800 need 80 hours or more. This seems strange since vicuna-13B costed much less. loss dropped to around 0.8, maybe a week later I shall report. Thanks for your debug again.
> 
> can you share the result on baichuan-13B? thx

on my private dataset, most are english, baichuan-13B's result is so worse
